core/org.wso2.carbon.tomcat.ext/src/main/java/org/wso2/carbon/tomcat/ext/transport/CarbonResponseWrapper.java

```diff
@@ -96,17 +97,24 @@ public String generateCookieString(Cookie cookie) {
             return this.getContext().getCookieProcessor().generateHeader(cookie, null);
         }
 
-        String cookieString = super.generateCookieString(cookie);
-        if (cookie instanceof ServletCookie) {
-            if (((ServletCookie) cookie).getSameSite() == null) {
-                cookieString = cookieString + "; SameSite=" + SameSiteCookie.STRICT.getName();
+        try {
+            String cookieString = super.generateCookieString(cookie);
+            if (cookie instanceof ServletCookie) {
+                if (((ServletCookie) cookie).getSameSite() == null) {
+                    cookieString = cookieString + "; SameSite=" + SameSiteCookie.STRICT.getName();
+                } else {
+                    cookieString = cookieString + "; SameSite=" + ((ServletCookie) cookie).getSameSite().getName();
+                }
             } else {
-                cookieString = cookieString + "; SameSite=" + ((ServletCookie) cookie).getSameSite().getName();
+                cookieString = cookieString + "; SameSite=" + SameSiteCookie.STRICT.getName();
             }
-        } else {
-            cookieString = cookieString + "; SameSite=" + SameSiteCookie.STRICT.getName();
+            return cookieString;
+        } catch (IllegalArgumentException e) {
```

Comment:
Shall we handle catching exception in `org.wso2.carbon.ui.CarbonUILoginUtil.saveOriginalUrl(CarbonUILoginUtil.java:123)`

Because the CarbonResponseWrapper is used in all the flows. So better not to hide the error during all of them. So let's catch this expectation within saveOriginalUrl addCookie method and log it as debug log and continue.

Reply:
The exception handling is moved to saveOriginalUrl method as suggested